Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -315,7 +315,9 @@ private func createResolvedPackages(
             }
         }
 
-        packageBuilder.dependencies = Array(dependencies.values)
+        packageBuilder.dependencies = dependencies.values
+            // Restore determinism:
+            .sorted(by: { $0.package.identity < $1.package.identity })
```

Comment:
Sorting these for stability is great, I think.  In the long run I think it would actually be nice if we could get back to using the order in which dependencies were declared in the manifest.  It shouldn't make a difference to the logic of course, but it seems to me that using the user's given order would be more natural to read as a package author.  It's a total detail, just something I hope that SwiftPM can get back to at some point (I think it used to do this early on).

Reply:
I thought about that too, and considered refactoring to use an `OrderedDictionary`. But pulling in `swift-collections` seemed too far off‐topic for this PR. It is certainly something to consider in the future.

Reply:
Absolutely.  It was more of a comment about a general approach here.  I should have made that clear.